frontend/src/serviceWorker.ts

```diff
@@ -0,0 +1,143 @@
+// This optional code is used to register a service worker.
```

Comment:
I know this file is optional, but you may want to consider testing the functions in here.

Reply:
Will do it as a part of the coming PRs.